In [1]:
# importomg necessary packages
import pandas as pd
from scipy import stats
import numpy as np

# Q_01)

In [2]:
df = pd.read_csv("Cutlets.csv")

### in this problem we can say

#### null hypothesis (H0) : unitA = unitB
#### Altenate hypothesis (H1) : unitA is not equal to unitB

In [3]:
df.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

### from the problem we have defined it will take two sample and two  tailed test

In [4]:
unitA = df.iloc[:,0]
unitA.head()

0    6.8090
1    6.4376
2    6.9157
3    7.3012
4    7.4488
Name: Unit A, dtype: float64

In [5]:
unitB = df.iloc[:,1]
unitB.head()

0    6.7703
1    7.5093
2    6.7300
3    6.7878
4    7.1522
Name: Unit B, dtype: float64

In [6]:
stats.ttest_ind(unitA,unitB)

Ttest_indResult(statistic=0.7228688704678063, pvalue=0.4722394724599501)

here we can say the p value is not signifiant to the alpha which we show as 0.4722 > 0.05
hence we do not reject the nullhypothesis 
means there is a no difference in between unitA and UnitB

# Q2)

In [7]:
ds = pd.read_csv("LabTAT.csv")

### in following scenario we can say here our null hypothesis is (H0) : average1 = average2 = average3
### and we can say here our Altenate hypothesis is (H1) : average1 not equal to average1 and average3 means there is a some diference

In [8]:
ds.head()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

In [9]:
Lab1 = ds.iloc[:,0]
Lab1.head()

0    185.35
1    170.49
2    192.77
3    177.33
4    193.41
Name: Laboratory 1, dtype: float64

In [10]:
Lab2 = ds.iloc[:,1]
Lab2.head()

0    165.53
1    185.91
2    194.92
3    183.00
4    169.57
Name: Laboratory 2, dtype: float64

In [11]:
Lab3 = ds.iloc[:,2]
Lab3.head()

0    176.70
1    198.45
2    201.23
3    199.61
4    204.63
Name: Laboratory 3, dtype: float64

In [12]:
Lab4 = ds.iloc[:,3]
Lab4.head()

0    166.13
1    160.79
2    185.18
3    176.42
4    152.60
Name: Laboratory 4, dtype: float64

In [13]:
stats.f_oneway(Lab1,Lab2,Lab3,Lab4)

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

here we can say the p value is signifiant to the alpha which we show as 2.1156708949992414e-57 < 0.05 hence we reject the nullhypothesis means there is a some difference in between Labroteries

# Q3)

In [14]:
dt = pd.read_csv("BuyerRatio.csv")

In [15]:
dt

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [16]:
sales = dt.iloc[:,1:5]

In [17]:
sales = sales.values

In [18]:
sales

array([[  50,  142,  131,   70],
       [ 435, 1523, 1356,  750]], dtype=int64)

In [19]:
stats.chi2_contingency(sales)

(1.595945538661058,
 0.6603094907091882,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

here we can say the p value is not signifiant to the alpha which we show as 0.6603094907091882 > 0.05 hence we do not reject the nullhypothesis 

# Q4)

In [20]:
dc = pd.read_csv("Costomer+OrderForm.csv")

In [21]:
dc.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [22]:
dc.Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [23]:
dc.Indonesia.value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [24]:
dc.Malta.value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [25]:
dc.India.value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [26]:
Telecall = np.array([[271,267,269,280],[29,33,31,20]])

In [27]:
stats.chi2_contingency(Telecall)

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

here we can say the p value is not signifiant to the alpha which we show as 0.2771020991233135 > 0.05 hence we do not reject the nullhypothesis 